swagger.yaml

```diff
@@ -1,75 +1,208 @@
+        - name: "id"
+          in: "path"
+          description: "Textbook ID"
+          required: true
+          type: "string"
+
       responses:
         200:
-          description: Object containing information
+          description: "Textbook successfully retrieved"
+          schema:
+            $ref: "#/definitions/TextbookResult"
+        404:
+          description: "Textbook not found"
           schema:
-            $ref: '#/definitions/Info'
+            $ref: "#/definitions/Error"
+        500:
+          description: "Internal server error"
+          schema:
+            $ref: "#/definitions/Error"
+
+  /textbooks:
+    get:
+      tags:
+      - "textbooks"
+      summary: "Retrieve textbooks by parameters"
+      operationId: "getTextbooks"
+      parameters:
+
+        - name: "term"
+          in: "query"
+          description: "Name of term in which to find textbooks"
+          type: "string"
+          required: true
+
+        - name: "department"
+          in: "query"
+          description: "Name of department in which to find textbooks"
+          type: "string"
+          required: true
 
-parameters:
-  pretty:
-    name: pretty
-    in: query
-    type: boolean
-    required: false
-    description: If true, JSON response will be pretty-printed
+        - name: "course"
+          in: "query"
+          description: "Name of course in which to find textbooks"
+          type: "string"
+          required: true
+
+        - name: "section"
+          in: "query"
+          description: "Name of section in which to find textbooks"
+          type: "string"
+          required: false
+
+        - name: "isRequired"
+          in: "query"
+          description: "If true, only retrieves required textbooks"
+          type: "string"
+          required: false
+          default: false
+
+      responses:
+        200:
+          description: "Textbooks successfully retrieved"
+          schema:
+            $ref: "#/definitions/TextbookResultArray"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+        500:
+          description: "Internal server error"
+          schema:
+            $ref: "#/definitions/Error"
 
 definitions:
-  Info:
+
+  Textbook:
+    type: "object"
     properties:
-      name:
-        type: string
-        description: Name of the API.
-      time:
-        type: string
-        description: Build time in ISO 8601 format.
-      unixTime:
-        type: integer
-        description: Build time as milliseconds since the unix epoch.
-      commit:
-        type: string
-        description: Git commit hash of the revision which was built.
-      documentation:
-        type: string
-        description: Filename of the swagger specification for the API.
+      isbn:
+        type: "string"
+        description: "ISBN of textbook"
+        example: "9780998819112"
+      required:
+        type: "string"
```

Comment:
Should this be a boolean instead of a string?

Reply:
verbacompare has a `required` field in each textbook. The only values I've seen so far are `"Required"` and `"Optional"` but I'm not sure if there might be others. Now that I think about it, I could just add logic in the api that explicitly checks for `"Required"` or `"Optional"` and if any other value exists, I can raise an exception or handle it some other way.